### DISCUSSION TITLE
Export public keys from private key set?

### DISCUSSION THREAD

**Discussion by bzuillsmith** [panva/jose#420] [Q&A]
Hey there. I'm updating an old app using v1. It looks like v1 let you create a JWKS from private keys and then you could call `.toJWKS(false)` to get a set with just the public keys. Is this possible in v4?

**Comment by treeder** [comment]
You figure this out?

**Reply by amitava82** [reply]
Anyone? I'm trying figure out how to generate jwks. Most examples I could find use node-jose

**Reply by treeder** [reply]
I figured out how to get the public key myself. What algorithm are you using? For Ed25519 it was pretty easy.

**Reply by amitava82** [reply]
RS256

**Reply by treeder** [reply]
For ed25519, I simply removed the "d" field from the JWK object.

**Comment by big-kahuna-burger** [comment]
When using JWK format with keys intended for use with asymmetric algorithms, the essential part is to understand that derivation of public key from private can simply be done by omitting properties that are related to private key only.
Take a look at examples here from the spec:
https://datatracker.ietf.org/doc/html/rfc7517#appendix-A

**Reply by treeder** [reply]
Ya, for ed25519, I simply removed the "d" field from the JWK object.

**Reply by amitava82** [reply]
I was able to get it working with following. I hope this is correct way to do it.
```
keys: [
      {
        ...keys.publicKey,
        alg: 'RS256',
        use: 'sig',
        kid: keys.kid,
      },
    ]
```    
Where publicKey is exported value from `exportJWK(publicKey)`
